httpclient/src/main/kotlin/tbdex/sdk/httpclient/RealTbdexClient.kt

```diff
@@ -0,0 +1,195 @@
+package tbdex.sdk.httpclient
+
+import com.fasterxml.jackson.module.kotlin.convertValue
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import tbdex.sdk.httpclient.Json.objectMapper
+import tbdex.sdk.httpclient.models.ErrorDetail
+import tbdex.sdk.httpclient.models.ErrorResponse
+import tbdex.sdk.httpclient.models.GetExchangeResponse
+import tbdex.sdk.httpclient.models.GetExchangesFilter
+import tbdex.sdk.httpclient.models.GetExchangesResponse
+import tbdex.sdk.httpclient.models.GetOfferingsFilter
+import tbdex.sdk.httpclient.models.GetOfferingsResponse
+import tbdex.sdk.httpclient.models.SendMessageResponse
+import tbdex.sdk.httpclient.models.TbdexResponse
+import tbdex.sdk.protocol.models.Message
+import tbdex.sdk.protocol.models.Offering
+import web5.sdk.dids.Did
+
+/**
+ * Real tbdex client
+ *
+ * @constructor Create empty Real tbdex client
+ */
+object RealTbdexClient : TbdexClient {
+  private val client = OkHttpClient()
+  private val jsonMediaType = "application/json; charset=utf-8".toMediaType()
+  private const val JSON_HEADER = "application/json"
+
+  override fun getOfferings(pfiDid: String, filter: GetOfferingsFilter?): TbdexResponse {
+    val pfiServiceEndpoint = getPfiServiceEndpoint(pfiDid)
+    val baseUrl = "$pfiServiceEndpoint/offerings/"
+
+    // compose query param
+    val queryMap: Map<String, String?>? = filter?.let { objectMapper.convertValue(it) }
+    val notNullQueryMap = queryMap?.filterValues { it != null }
+    val httpUrlBuilder = baseUrl.toHttpUrl().newBuilder()
+    notNullQueryMap?.forEach { httpUrlBuilder.addQueryParameter(it.key, it.value) }
+
+    val request = Request.Builder()
+      .url(httpUrlBuilder.build())
+      .get()
+      .build()
+
+    val response: Response = client.newCall(request).execute()
+    return when {
+      response.isSuccessful -> {
+        val responseString = response.body?.string()
+        val jsonNode = objectMapper.readTree(responseString)
+        val jsonData = jsonNode.get("data").asText()
```

Comment:
Running into issues b/c `asText()` is returning an empty string. We could do `.toString()`.

May be simpler to just do

```kotlin
val data = objectMapper.readValue(response.body?.string(), GetOfferingsResponse::class.java).data
```

But this, as well as the `toString()` is throwing a deserialization exception

```
/**
         * Exception caught in remittances API: Cannot construct instance of `org.everit.json.schema.Schema` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
         *  at [Source: (String)"{"data":[{"metadata":{"from":"did:ion:EiC_tpTAKoHLW2Z2CJ9plAMbFAOoWR8CgAqQzCdADcL5TQ:eyJkZWx0YSI6eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJyZXBsYWNlIiwiZG9jdW1lbnQiOnsicHVibGljS2V5cyI6W3siaWQiOiJkd24tc2lnIiwicHVibGljS2V5SndrIjp7ImNydiI6IkVkMjU1MTkiLCJrdHkiOiJPS1AiLCJ4IjoiSW1MSnRDT01CTE5INHF5aXVrdVFIdlZobkE2OGtZZTU2V0RlY214RzBGWSJ9LCJwdXJwb3NlcyI6WyJhdXRoZW50aWNhdGlvbiJdLCJ0eXBlIjoiSnNvbldlYktleTIwMjAifV0sInNlcnZpY2VzIjpbeyJpZCI6InBmaSIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTAwMCIsInR5cGUiOiJQRkk"[truncated 3237 chars]; line: 1, column: 1112] (through reference chain: tbdex.sdk.httpclient.models.GetOfferingsResponse["data"]->java.util.ArrayList[0]->tbdex.sdk.protocol.models.Offering["data"]->tbdex.sdk.protocol.models.OfferingData["payinMethods"]->java.util.ArrayList[0]->tbdex.sdk.protocol.models.PaymentMethod["requiredPaymentDetails"])
         * com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.everit.json.schema.Schema` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
         *  at [Source: (String)"{"data":[{"metadata":{"from":"did:ion:EiC_tpTAKoHLW2Z2CJ9plAMbFAOoWR8CgAqQzCdADcL5TQ
         */
```

Looking into it...

Reply:
So the issue is `Schema` is an abstract class. I think we should be using `ObjectSchema` instead, but now running into a regex exception

```
Exception caught in remittances API: Cannot find a (Map) Key deserializer for type [simple type, class org.everit.json.schema.regexp.Regexp]
 at [Source: (String)"{"data":[{"metadata":{"from":"did:ion:EiC_tpTAKoHLW2Z2CJ9plAMbFAOoWR8CgAqQzCdADcL5TQ:eyJkZWx0YSI6eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJyZXBsYWNlIiwiZG9jdW1lbnQiOnsicHVibGljS2V5cyI6W3siaWQiOiJkd24tc2lnIiwicHVibGljS2V5SndrIjp7ImNydiI6IkVkMjU1MTkiLCJrdHkiOiJPS1AiLCJ4IjoiSW1MSnRDT01CTE5INHF5aXVrdVFIdlZobkE2OGtZZTU2V0RlY214RzBGWSJ9LCJwdXJwb3NlcyI6WyJhdXRoZW50aWNhdGlvbiJdLCJ0eXBlIjoiSnNvbldlYktleTIwMjAifV0sInNlcnZpY2VzIjpbeyJpZCI6InBmaSIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTAwMCIsInR5cGUiOiJQRkk"[truncated 3237 chars]; line: 1, column: 1]
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot find a (Map) Key deserializer for type [simple type, class org.everit.json.schema.regexp.Regexp]
 at [Source: (String)"{"data":[{"metadata":{"from":"did:ion:EiC_tpTAKoHLW2Z2CJ9plAMbFAOoWR8CgAqQzCdADcL5TQ:eyJkZWx0YSI6eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJyZXBsYWNlIiwiZG9jdW1lbnQiOnsicHVibGljS2V5cyI6W3siaWQiOiJkd24tc2lnIiwicHVibGljS2V5SndrIjp7ImNydiI6IkVkMjU1MTkiLCJrdHkiOiJPS1AiLCJ4IjoiSW1MSnRDT01CTE5INHF5aXVrdVFIdlZobkE2OGtZZTU2V0RlY214RzBGWSJ9LCJwdXJwb3NlcyI6WyJhdXRoZW50aWNhdGlvbiJdLCJ0eXBlIjoiSnNvbldlYktleTIwMjAifV0sInNlcnZpY2VzIjpbeyJpZCI6InBmaSIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTAwMCIsInR5cGUiOiJQRkk"[truncated 3237 chars]; line: 1, column: 1]
```

Reply:
Okay being told we're migrating off everit, we'll need to do that first then fix these issues

Reply:
Ah that's a bug, we should just use strings for that field like we do with claims. Thanks!

Reply:
`requiredPaymentDetails` should be a string? From the mock PFI it's coming back as an object

```json
"payoutMethods": [
	{
		"kind": "MOMO_MPESA",
		"requiredPaymentDetails": {
			"$schema": "http://json-schema.org/draft-07/schema#",
			"title": "Mobile Money Required Payment Details",
			"type": "object",
			"required": [
				"phoneNumber",
				"reason"
			],
			"additionalProperties": false,
			"properties": {
				"phoneNumber": {
					"title": "Mobile money phone number",
					"description": "Phone number of the Mobile Money account",
					"type": "string"
				},
				"reason": {
					"title": "Reason for sending",
					"description": "To abide by the travel rules and financial reporting requirements, the reason for sending money",
					"type": "string"
				}
			}
		}
```

Reply:
It's a json schema in the js SDK. If `networknt` gives us a schema type, could use that, otherwise assume that it's a string and then parse it using `networknt` when validating the rfq/offering